js/ui/screenShield.js

```diff
@@ -580,6 +580,47 @@ var ScreenShield = new Lang.Class({
         this._cursorTracker = Meta.CursorTracker.get_for_screen(global.screen);
 
         this._syncInhibitor();
+
+        Main.paygManager.connect('code-expired', () => { this.lock(true) });
+        Main.paygManager.connect('expiry-time-changed', () => {
+            let userManager = AccountsService.UserManager.get_default();
+            let user = userManager.get_user(GLib.get_user_name());
+
+            // A new valid code has been introduced but the machine is
+            // not unlocked yet -> Make sure we stay in the locked state.
```

Comment:
This shouldn’t happen, I think (whenever a code is successfully verified, the `ExpiryTime` advertised by the daemon is set to now + the period of the code, so it should always be in the future). Is this branch just as a safety net?

Reply:
It is just that, yes